design/ceph-mixins.md

```diff
@@ -0,0 +1,61 @@
+# Monitoring ceph storage with ceph-mixins
+
+---
+
+## Overview
+
+Currently, rook deploys ceph storage but not ceph monitoring. It can be deployed manually using documented steps. But still, it lacks Prometheus Alerts and Recording Rules that is useful in easy monitoring of ceph storage.
+
+This is where **ceph-mixins** comes in. Ceph-mixins defines a minimalistic approach to package together the ceph storage specific prometheus alerts, recording rules and grafana dashboards, in a simple and platform agnostic way.
+
+Rook can use ceph-mixins as the default solution for monitoring ceph storage.
+
+## Design
+
+---
+
+### Responsibilities
+
+1. Getting the latest updates from ceph-mixins into rook.
+2. Building ceph-mixins artifacts into kubernetes custom resources.
+3. Deploying prometheus bundle generated by ceph-mixins.
+
+### Deployment
+
+1. Fetching latest resources from ceph-mixins
+
+   Ceph-mixins uses Jsonnet for writing very flexible and extensible configurations pertaining to prometheus alerts, recording rules and grafana dashboards.
+
+   These configuration files are bundled together and managed using jsonnet-bundler.
+
+    **Changes required in Rook**
+
+    A new `jsonnet` directory should be created in Rook which will hold configurations which will be used to pull resources from ceph-mixins. These configurations will allow rook to version lock the resources being pulled and extend or modify it.
+
+2. Building deployment assets
+
+   A build script will be used to generate yaml files that can be applied to any kubernetes cluster.
```

Comment:
Will multiple yaml files be generated? Below, it looks like there could be a single `prometheus-rules.yaml` file.

Reply:
A single yaml file will be checked-in to accomodate all the alerting and recording rules. (Deployment of Prometheus stack will be left to the user.)